apps/desktop/src/lib/stacks/stacksService.ts

```diff
@@ -0,0 +1,50 @@
+import { reduxApi } from '$lib/redux/api';
+import { DesktopRedux } from '$lib/redux/store.svelte';
+import type { Unsubscribe } from '$lib/redux/unsubscribe';
+import type { Stack } from './stack';
+
+/** Poll every five minutes to mitigate potential bugs. */
+const POLLING_INTERVAL = 5 * 60 * 1000;
+
+enum Tags {
+	Stacks = 'Stacks'
+}
+
+export class StacksService {
+	private stacksApi = reduxApi.injectEndpoints({
+		endpoints: (build) => ({
+			/** Fetches all stacks. */
+			get: build.query<Stack[], { projectId: string }>({
+				query: ({ projectId }) => ({ command: 'stacks', params: { projectId } }),
+				providesTags: [Tags.Stacks]
+			}),
+			new: build.mutation<Stack, { projectId: string }>({
+				query: ({ projectId }) => ({
+					command: 'create_virtual_branch',
+					params: { projectId, branch: {} }
+				}),
+				invalidatesTags: [Tags.Stacks]
+			})
+		})
+	});
+
+	constructor(private state: DesktopRedux) {}
+
+	poll(projectId: string): Unsubscribe {
+		return this.state.dispatch(
+			this.stacksApi.endpoints.get.initiate(
+				{ projectId },
+				{ subscribe: true, subscriptionOptions: { pollingInterval: POLLING_INTERVAL } }
+			)
+		).unsubscribe;
+	}
+
+	select(projectId: string) {
+		return this.stacksApi.endpoints.get.select({ projectId });
+	}
+
+	// eslint-disable-next-line @typescript-eslint/promise-function-async
+	new(projectId: string) {
+		return this.state.dispatch(this.stacksApi.endpoints.new.initiate({ projectId }));
+	}
```

Comment:
Redux returns promise like objects that we do not want to automatically await. The return value of `initiate` has additional methods like `abort`, `unsubscribe` etc. Should we remove this lint rule or disable it on a case-by-case basis?